mlir/lib/Dialect/Vector/Transforms/LowerVectorInterleave.cpp

```diff
@@ -79,6 +79,65 @@ class UnrollInterleaveOp final : public OpRewritePattern<vector::InterleaveOp> {
   int64_t targetRank = 1;
 };
 
+/// A one-shot unrolling of vector.deinterleave to the `targetRank`.
+///
+/// Example:
+///
+/// ```mlir
+/// vector.deinterleave %a : vector<1x2x3x8xi64> -> vector<1x2x3x4xi64>
+/// ```
+/// Would be unrolled to:
+/// ```mlir
+/// %result = arith.constant dense<0> : vector<1x2x3x8xi64>
+/// %0 = vector.extract %a[0, 0, 0]                 ─┐
+///        : vector<4xi64> from vector<1x2x3x4xi64>  |  - Repeated 6x for
+/// %1, %2 = vector.deinterleave %0 :                |    all leading positions
+///        : vector<8xi64> -> vector<4xi64>          |
+/// %3 = vector.insert %1, %result [0, 0, 0]         |
+///        : vector<4xi64> into vector<1x2x3x4xi64>  |
+/// %3 = vector.insert %2, %result [0, 0, 0]         |
+///        : vector<4xi64> into vector<1x2x3x4xi64>  ┘
```

Comment:
This example does not look correct for deinterleave. There's one result and both the even/odd parts results of the deinterleave are going in the same result, and the types don't look correct either.

Reply:
The single `%result` is fine (misread this), but the type of the result is not correct.

Reply:
The result is `vector<1x2x3x8xi64>`, but the deinterleave seems to be producing `vector<4xi64>`, but the extract is also of `vector<4xi64>` :sweat_smile: I think the types are very mixed up here.